File: src/java/org/elastos/hive/vault/scripting/RegScriptParams.java
package org.elastos.hive.vault.scripting;

import com.google.gson.annotations.SerializedName;

class RegScriptParams {
	@SerializedName("executable")
	private Executable executable;
	@SerializedName("allowAnonymousUser")
	private Boolean allowAnonymousUser;
	@SerializedName("allowAnonymousApp")
	private Boolean allowAnonymousApp;
	@SerializedName("condition")
	private Condition condition;

	public RegScriptParams setExecutable(Executable executable) {
		this.executable = executable;
		return this;
	}

	public RegScriptParams setAllowAnonymousUser(Boolean allowAnonymousUser) {
		this.allowAnonymousUser = allowAnonymousUser;
		return this;
	}

	public RegScriptParams setAllowAnonymousApp(Boolean allowAnonymousApp) {
		this.allowAnonymousApp = allowAnonymousApp;
		return this;
	}

	public RegScriptParams setCondition(Condition condition) {
		this.condition = condition;
		return this;
	}
}

File: src/restclient/scripting/RegScriptParams.ts
import { Condition } from "./condition";
import { Executable } from "./executable";

export class RegScriptParams {
	private executable: Executable;
	private allowAnonymousUser: Boolean;
	private allowAnonymousApp: Boolean;
	private condition: Condition;

	setExecutable(executable: Executable) : RegScriptParams {
		this.executable = executable;
		return this;
	}

	setAllowAnonymousUser(allowAnonymousUser: boolean) : RegScriptParams{
		this.allowAnonymousUser = allowAnonymousUser;
		return this;
	}

	setAllowAnonymousApp(allowAnonymousApp:boolean) : RegScriptParams{
		this.allowAnonymousApp = allowAnonymousApp;
		return this;
	}

	setCondition( condition: Condition) : RegScriptParams{
		this.condition = condition;
		return this;
	}

	toString() : string {
		return JSON.stringify(this);
	}
}
